web/src/components/Broadcast.js

```diff
@@ -0,0 +1,52 @@
+import { useEffect } from 'react';
+
+let channel = null;
+const handlers = [];
+
+const postMessage = msg => {
+  if (channel) {
+    channel.postMessage(msg);
+  }
+};
+
+const subscribe = fn => {
+  if (handlers.indexOf(fn) < 0) {
+    handlers.push(fn);
+  }
+};
+
+const unsubscribe = fn => {
+  const index = handlers.indexOf(fn);
+  if (index >= 0) {
+    handlers.splice(index, 1);
+  }
+};
+
+const Broadcast = () => {
+  useEffect(() => {
+    try {
+      if (window.BroadcastChannel) {
+        channel = new BroadcastChannel('eapd_activity');
+        channel.onmessage = ({ isTrusted, data }) => {
+          if (isTrusted) {
```

Comment:
The DOM [`Event.isTrusted` property](https://developer.mozilla.org/en-US/docs/Web/API/Event/isTrusted) is cool because it indicates a user action rather than a scripted action. I'm not sure that's what we want for a *generic* broadcast handler, but it's what we want for this particular PR.

It might be worth revisiting at some point, though - how neat would it be if you accidentally open multiple tabs of eAPD and then they all stay in sync because when you update one, it broadcasts the update to the other tabs?

![pretty cool](https://media.giphy.com/media/3ohhwFwSstH36WVQZi/giphy.gif)